### PR TITLE
allow caching in sortable set and use it

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
refactor caching

**Did you add tests for your changes?**
existing tests

**If relevant, link to documentation update:**
N/A

**Summary**
more caching -> performance

**Does this PR introduce a breaking change?**
yes

**Other information**

### CODE REVIEW

File: lib/Chunk.js [hunk 1]
```diff
@@ -21,6 +21,27 @@ const sortByIdentifier = (a, b) => {
...
+const getModulesIdent = set => {
+	set.sort();
+	let str = "";
+	set.forEach(m => {
+		str += m.identifier() + "#";
+	});
+	return str;
+};
+
+const getArray = set => Array.from(set);
```
Comment (michael-ciniawsky) [inline lib/Chunk.js]: move to `lib/util/toArray.js` && `getArray` =>`toArray` ?

File: lib/Chunk.js [hunk 2]
```diff
@@ -21,6 +21,27 @@ const sortByIdentifier = (a, b) => {
 	return 0;
 };
 
+const getFrozenArray = set => Object.freeze(Array.from(this));
```
Comment (michael-ciniawsky) [inline lib/Chunk.js]: move to `lib/util/toFrozenArray.js` && `getFrozenArray` =>`toFrozenArray` ?

Comment (michael-ciniawsky) [inline lib/Chunk.js]: move to `lib/util/toFrozenArray.js` && `getFrozenArray` => `toFrozenArray` ?

File: lib/Chunk.js (hunk 1 above)
Comment (michael-ciniawsky) [inline lib/Chunk.js]: move to `lib/util/toArray.js` && `getArray` => `toArray` ?

File: lib/Chunk.js [hunk 3]
```diff
@@ -65,7 +86,7 @@ class Chunk {
 	 * @return {Array} - an array containing the chunks
 	 */
 	getChunks() {
-		return Array.from(this._chunks);
+		return this._chunks.getCachedInfo(getArray);
```
Comment (michael-ciniawsky) [inline lib/Chunk.js]: `getCache(d)` ?

File: lib/Module.js [hunk 4]
```diff
@@ -21,6 +21,25 @@ const sortByDebugId = (a, b) => {
 	return a.debugId - b.debugId;
 };
 
+const getFrozenArray = set => Object.freeze(Array.from(this));
```
Comment (michael-ciniawsky) [inline lib/Module.js]: use `lib/util/toFrozenArray.js` ? :)

File: lib/Chunk.js [hunk 5]
```diff
@@ -269,16 +290,11 @@ class Chunk {
...
 	}
 
 	getModulesIdent() {
-		this._modules.sort();
-		let str = "";
-		this._modules.forEach(m => {
-			str += m.identifier() + "#";
-		});
-		return str;
+		return this._modules.getOrderIndependentCachedInfo(getModulesIdent);
```
Comment (michael-ciniawsky) [inline lib/Chunk.js]: `getOrderIndependentCache(d)` 

> OrderIndenpendent => Unordered/Unsorted/Orderless

File: lib/Chunk.js (hunk 1 above)
Comment (sokra) [reply lib/Chunk.js]: I don't think a separate module make sense for a single line of code. This increases code size, decreases performance and readablility.

Comment (michael-ciniawsky) [reply lib/Chunk.js]: Yeah, it isn't _really_ needed here, especially after the last refactor :)

**Pull request merged**